api4/system.go

```diff
@@ -7,15 +7,53 @@ import (
 	"net/http"
 
 	l4g "github.com/alecthomas/log4go"
+	"github.com/mattermost/platform/app"
+	"github.com/mattermost/platform/model"
 	"github.com/mattermost/platform/utils"
 )
 
 func InitSystem() {
 	l4g.Debug(utils.T("api.system.init.debug"))
 
 	BaseRoutes.System.Handle("/ping", ApiHandler(getSystemPing)).Methods("GET")
+	BaseRoutes.ApiRoot.Handle("/config", ApiSessionRequired(updateConfig)).Methods("PUT")
+
 }
 
 func getSystemPing(c *Context, w http.ResponseWriter, r *http.Request) {
 	ReturnStatusOK(w)
 }
+
+func updateConfig(c *Context, w http.ResponseWriter, r *http.Request) {
+	cfg := model.ConfigFromJson(r.Body)
+	if cfg == nil {
+		c.SetInvalidParam("config")
+		return
+	}
+
+	cfg.SetDefaults()
+
+	err := cfg.IsValid()
+	if err != nil {
+		c.Err = err
+		return
+	}
```

Comment:
I tested the endpoint sending a configuration file with missing sections and I got panic
with this it works.

But I can make the test again and check it